### DISCUSSION TITLE
Is there a demo about MiMC-p/p encryption proof by gnark?

### DISCUSSION THREAD

**Discussion by iamzhanghaohui** [Consensys/gnark#628] [Q&A] [answered]
I have read about the application of the MiMC-p/p in encryption and passing zero-knowledge proofs, and I am curious if the gnark library can support this functionality. I would appreciate if someone could clarify this matter for me. Thank you.

I apologize if my question appears imprecise, and I appreciate any help or guidance that anyone is able to provide. Thank you for your time and assistance.

**Comment by ivokub** [comment, marked as answer]
gnark has MiMC hasher gadget: https://github.com/ConsenSys/gnark/tree/master/std/hash/mimc

I hope this works.
